src/base/bittorrent/session.cpp

```diff
@@ -417,6 +417,13 @@ Session::Session(QObject *parent)
     , m_utpMixedMode(BITTORRENT_SESSION_KEY("uTPMixedMode"), MixedModeAlgorithm::TCP
         , clampValue(MixedModeAlgorithm::TCP, MixedModeAlgorithm::Proportional))
     , m_multiConnectionsPerIpEnabled(BITTORRENT_SESSION_KEY("MultiConnectionsPerIp"), false)
+    , m_validateHTTPSTrackerCertificate(BITTORRENT_SESSION_KEY("ValidateHTTPSTrackerCertificate"), false)
+    , m_httpsTrackerTLSVersion(BITTORRENT_SESSION_KEY("HTTPSTrackerTLSVersion"), TLSVersion::TLS12
+#if (TORRENT_USE_TLS13 == 1)
```

Comment:
libtorrent always defines `TORRENT_USE_TLS13`, sometimes to 0 and sometimes to 1. The benefit of this over a macro that is only sometimes defined is that `#if MACRO` will cause a warning if `MACRO` is not defined, i.e. it will catch typos.

Reply:
>`#if MACRO` will cause a warning if MACRO is not defined,

There is `#ifdef` for macro testing. Why one should use `#if` for this purpose?

Reply:
https://github.com/arvidn/libtorrent/pull/4415/files#diff-f04aa2203ab9fbc2523e1afeb7e96a26R571
For older libtorrent it will be undefined and `0` for using boost < 1.69 and `1` otherwise.

Reply:
> There is #ifdef for macro testing. Why one should use #if for this purpose?

To avoid bugs where the name of the macro is misspelled.

> 
For older libtorrent version it will be undefined and 0 for using boost < 1.69 and 1 otherwise.

That's a good point. This test probably need to be:

```
#if defined TORRENT_USE_TLS13 && TORRENT_USE_TLS13
```

(the existing code works, but I would imagine it generates a warning, if qbt is built with warnings enabled).

Reply:
>(the existing code works, but I would imagine it generates a warning, if qbt is built with warnings enabled).

No new warnings found in CI builds, seems we are fine here.